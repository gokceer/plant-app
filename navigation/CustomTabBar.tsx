import ScanIcon from "@/assets/icons/ScanIcon";
import { useRouter } from "expo-router";
import React from "react";
import { Pressable, StyleSheet, View } from "react-native";

const BUTTON_WIDTH = 60;

const CustomTabBar = () => {
  const router = useRouter();
  return (
    <Pressable
      onPress={() => router.push("/scan")}
      style={styles.scanButtonContainer}
      testID="customTabBarButton"
    >
      <View style={styles.scanButton}>
        <ScanIcon />
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  scanButtonContainer: {
    position: "absolute",
    bottom: 10,
    left: "50%",
    transform: [{ translateX: -BUTTON_WIDTH / 2 }],
  },
  scanButton: {
    width: BUTTON_WIDTH,
    height: 60,
    borderRadius: 30,
    backgroundColor: "#28AF6E",
    alignItems: "center",
    justifyContent: "center",
    elevation: 4,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.2,
    shadowRadius: 6,
    borderWidth: 4,
    borderColor: "#2CCC80",
  },
});

export default CustomTabBar;
