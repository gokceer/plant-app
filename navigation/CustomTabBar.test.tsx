import { fireEvent, render } from "@testing-library/react-native";
import { useRouter } from "expo-router";
import React from "react";
import CustomTabBar from "./CustomTabBar";

jest.mock("expo-router", () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn(),
  })),
}));

describe("CustomTabBar Component", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<CustomTabBar />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("calls onPress when pressed", () => {
    const mockPush = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({ push: mockPush });

    const { getByTestId } = render(<CustomTabBar />);
    fireEvent.press(getByTestId("customTabBarButton"));

    expect(mockPush).toHaveBeenCalledWith("/scan");
  });
});
