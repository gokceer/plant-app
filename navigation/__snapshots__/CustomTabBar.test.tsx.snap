// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTabBar Component renders correctly and matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "bottom": 10,
      "left": "50%",
      "position": "absolute",
      "transform": [
        {
          "translateX": -30,
        },
      ],
    }
  }
  testID="customTabBarButton"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#28AF6E",
        "borderColor": "#2CCC80",
        "borderRadius": 30,
        "borderWidth": 4,
        "elevation": 4,
        "height": 60,
        "justifyContent": "center",
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 6,
        "width": 60,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={25}
      bbWidth={25}
      fill="none"
      focusable={false}
      height={25}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      vbHeight={25}
      vbWidth={25}
      width={25}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M8.788 1.786c.446 0 .809.365.809.815 0 .45-.362.814-.81.814l-1.73.002A2.623 2.623 0 004.45 6.045v2.029c0 .45-.364.815-.81.815a.813.813 0 01-.809-.815v-2.03c0-2.345 1.895-4.255 4.224-4.257l1.732-.001zm7.624 0h1.693c2.335 0 4.234 1.911 4.234 4.262v2.026c0 .45-.362.815-.81.815a.813.813 0 01-.808-.815V6.048a2.628 2.628 0 00-2.616-2.633h-1.693a.811.811 0 01-.809-.814c0-.45.363-.815.809-.815zm-.89 4.084H9.65a2.445 2.445 0 00-2.412 2.473v1.359a.3.3 0 00.287.293h10.119a.301.301 0 00.29-.293V8.343a2.508 2.508 0 00-.703-1.75 2.384 2.384 0 00-1.71-.723zm-13.82 5.782h21.766c.447 0 .81.365.81.816 0 .449-.363.813-.81.813H22.34v3.884c0 2.352-1.9 4.264-4.234 4.264h-1.693a.813.813 0 01-.81-.816c0-.45.362-.814.81-.814h1.693a2.627 2.627 0 002.616-2.634v-3.884h-2.788v1.082a2.445 2.445 0 01-2.413 2.473H9.649a2.445 2.445 0 01-2.412-2.473V13.28H4.45v3.889a2.624 2.624 0 002.609 2.628h1.73c.447 0 .81.366.81.815a.813.813 0 01-.81.816l-1.732-.002c-2.33-.002-4.225-1.912-4.225-4.257v-3.89H1.703a.812.812 0 01-.81-.813c0-.45.363-.815.81-.815z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
